include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -573,6 +662,26 @@ class __future : private std::tuple<_Args...>
     }
 };
 
+// Only use USM host allocations on L0 GPUs. Other devices show significant slowdowns and will use a buffer instead.
+inline bool
+__use_USM_host_allocations(sycl::queue __queue)
+{
+// The unified future above is currently only supported by DPCPP 2023.1 and compilers starting from 20230820.
```

Comment:
It looks very weird...
Why would not just call `__device.has(sycl::aspect::usm_host_allocations`? or why such call doesn't  make the all necessary checks?
A question rather is to a runtime team...

Reply:
We can query the support for USM host memory with `sycl::aspect::usm_host_allocations`. However, its use has shown to be beneficial on L0 GPUs only. This routine checks for all three conditions and uses guards to fall back to USM for compilers not supporting the unified future implementation.